infrastructure/src/test/java/org/corfudb/infrastructure/SequencerServerTest.java

```diff
@@ -0,0 +1,898 @@
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.roaringbitmap.longlong.Roaring64NavigableMap;
+
+import java.io.ObjectInputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.corfudb.protocols.CorfuProtocolCommon.DEFAULT_UUID;
+import static org.corfudb.protocols.CorfuProtocolCommon.getStreamAddressSpace;
+import static org.corfudb.protocols.CorfuProtocolCommon.getUuidMsg;
+import static org.corfudb.protocols.service.CorfuProtocolMessage.getHeaderMsg;
+import static org.corfudb.protocols.service.CorfuProtocolMessage.getRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolSequencer.getAllStreamsAddressRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolSequencer.getBootstrapSequencerRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolSequencer.getDefaultSequencerMetricsRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolSequencer.getSequencerTrimRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolSequencer.getStreamsAddressRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolSequencer.getTokenRequestMsg;
+import static org.corfudb.protocols.service.CorfuProtocolSequencer.getTokenResponse;
+import static org.corfudb.runtime.proto.RpcCommon.SequencerMetricsMsg.SequencerStatus;
+import static org.corfudb.runtime.proto.service.Sequencer.BootstrapSequencerRequestMsg;
+import static org.corfudb.runtime.proto.service.Sequencer.BootstrapSequencerResponseMsg;
+import static org.corfudb.runtime.proto.service.Sequencer.StreamsAddressRequestMsg;
+import static org.corfudb.runtime.proto.service.Sequencer.TokenRequestMsg;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * This class has test methods that test the RPC calls present in {@link SequencerServer}.
+ * The RequestMsg type messages that are tested are as follows -
+ * - TokenRequestMsg
+ * - BootstrapSequencerRequestMsg
+ * - SequencerTrimRequestMsg
+ * - SequencerMetricsRequestMsg
+ * - StreamsAddressRequestMsg
+ * See test methods for their functionality descriptions.
+ */
+@Slf4j
+public class SequencerServerTest {
```

Comment:
Hmm, I thought it was Maithem's suggestion to keep this name (we originally name them like CorfuSequencerServerTest, etc). They might not be able to be combined since these two test sets have quite different setup, and we intended to replace the old ones with the new Mockito unit tests later. Maybe @zfrenette and @Maithem could add more detail on this?